src/Variable.cpp

```diff
@@ -742,6 +755,9 @@ Variable::Output(std::ostream &out) const
 		assert(CGOptions::access_once() && "access_once is disabled!");
 		out << "ACCESS_ONCE(" << get_actual_name() << ")";
 	}
+	else if (isVector){
+		out << get_actual_name() << "[0]";
```

Comment:
Does it mean you always access the first element of a vector type? Why not a random element?

Reply:
Yes.
I tried doing `"[" << rnd_flipcoin(num_elements) << "]"` but I get segmentation fault - floating point exception (couldn't figure out why it is happening). But yes agree with you, correct approach is to select random element.

Reply:
Then it's probably an access that is out of array. Have you tried `valgrind` or sanitizers:

```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3d0f17e..a9ab258 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -121,7 +121,7 @@ add_definitions("-DHAVE_CONFIG_H")
 
 if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU"
     OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wno-long-long")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wno-long-long -fsanitize=address,undefined")
   set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3")
 endif()
 
```

Reply:
Got it after some debugging with sanitize option. My bad. The function `rnd_flipcoin()` does not accept 0 and hence led us to segmentation fault.

(Now preparing for push from all your above suggested changes)

Reply:
Here, I am able to use `"[" << rnd_flipcoin(num_elements) << "]"` in the place of `"[0]"` now.

Now, I am left with one error to fix i.e assigning vector elements addresses to pointer variables (which is of course not correct).